ResumableUpload.CS.Sample/ResumeableUploadSample.cs

```diff
@@ -0,0 +1,241 @@
+        /// 
+        /// See https://developers.google.com/api-client-library/dotnet/guide/media_upload for more details regarding ResumableUpload.
+        /// See https://developers.google.com/resources/api-libraries/documentation/youtube/v3/csharp/latest/ for YouTube Data API documentation.
+        /// See https://developers.google.com/youtube/v3/getting-started for details of the underlying API calls used by the .NET Client Library.
+        /// See https://github.com/google/google-api-dotnet-client/releases for .NET Client Library Release Notes.
+        /// See https://www.nuget.org/packages/Google.Apis.YouTube.v3/ for information on the .NET Client Library NuGet package.
+        /// 
+        /// Prepare for running the sample program:
+        ///   1. Add Reference to System.Net.Http
+        ///   2. In NuGet Package Manager Console, enter:
+        ///       Install-Package Google.Apis.YouTube.v3 
+        ///   3. Click Tools / NuGet Package Manager / Manage NuGet Packages for Solution
+        ///   4. Click on Updates / nuget.org and install all update packages
+        ///   5. Insert your ClientId and ClientSecret in the source code below.
+        ///   6. Change strFilePath below to specify the test video file to be uploaded.
+        /// </summary>
+        const int MAX_RESUME_RETRIES = 3;
+        static VideosResource.InsertMediaUpload objVideosInsertRequest = null;
+        static System.Threading.CancellationToken objCancellationToken;
+
+        static void Main(string[] args)
+        {
+            ExecuteUploadProcess().Wait();
+        }
+
+        /// <summary>
+        /// Execute the upload of one file to YouTube.
+        /// </summary>
+        private static async Task ExecuteUploadProcess()
+        {
+            const int intMinimumChunkSize = 256 * 1024; // .25MB (256K)
+            const int intHalfMBChunkSize = intMinimumChunkSize * 2; // .5MB (512K)
+            const int HTTP_CLIENT_TIMEOUT_MINUTES = 2;
+            UserCredential objOAUth2Credential = null;
+            YouTubeService objYouTubeService = null;
+            bool bOKtoContinue = false;
+            try
+             {
+            //
+            // ClientId and ClientSecret are found in your client_secret_*****.apps.googleusercontent.com.json file downloaded from 
+            // the Google Developers Console ( https://console.developers.google.com). 
+            // This sample shows the ClientID and ClientSecret in the source code. 
+            // Other samples in the sample library show how to read the Client Secrets from the client_secret_*****.apps.googleusercontent.com.json file. 
+            //
+            objOAUth2Credential = await GoogleWebAuthorizationBroker.AuthorizeAsync (
+                 new ClientSecrets {ClientId = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.apps.googleusercontent.com",
+                                    ClientSecret = "xxxxxxxxxxxxxxxxxxxxxxxx"},
```

Comment:
Rather than having developers scroll down to line 68 to enter their credentials, perhaps you could create a constant class field `clientId` and just reference that here? IMO that would be easier to deal with, but feel free to ignore.